api/src/main/java/org/openmrs/api/impl/ProviderServiceImpl.java

```diff
@@ -293,4 +294,17 @@ public Provider getUnknownProvider() {
 		return getProviderByUuid(Context.getAdministrationService().getGlobalProperty(
 		    OpenmrsConstants.GP_UNKNOWN_PROVIDER_UUID));
 	}
+
+	@Override
+	public Provider createProviderFromUser(User user) {
+		Provider p = new Provider();
+		if (user == null || user.getPerson() == null) {
+			throw new RuntimeException("User can not be null");
+		} else if (Context.getProviderService().getProvidersByPerson(user.getPerson()).isEmpty()) {
+			p.setPerson(user.getPerson());
+		} else {
+			throw new RuntimeException("Unable to create Provider from user");
+		}
+		return Context.getProviderService().saveProvider(p);
```

Comment:
Then, to go along with the suggestion above, the "saveProvider" call would have to moved up into the code block where we p.setPerson(user.getPerson())

Reply:
(And then I'd create another test case to test this scenario)